src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -146,14 +149,20 @@ internal class HttpManager : IHttpManager
             return response;
         }
 
-        private HttpClient GetHttpClient(X509Certificate2 x509Certificate2)
+        private HttpClient GetHttpClient(X509Certificate2 x509Certificate2, Func<HttpClient> validateServerCertificateCallback)
         {
-            if (_httpClientFactory is IMsalMtlsHttpClientFactory msalMtlsHttpClientFactory)
+            if (x509Certificate2 != null && _httpClientFactory is IMsalMtlsHttpClientFactory msalMtlsHttpClientFactory)
             {
                 // If the factory is an IMsalMtlsHttpClientFactory, use it to get an HttpClient with the certificate
                 return msalMtlsHttpClientFactory.GetHttpClient(x509Certificate2);
             }
 
+            HttpClient client = validateServerCertificateCallback?.Invoke();
```

Comment:
I think you should throw an "unssuported exception" or at least add a Debug.Assert if x509certificate2 is used in combination with this custom httpclient.

The custom httpclient should take precedence.

Reply:
I think I discussed something on these lines with @GladwinJohnson and my understanding is that if the x509Certificate2 is there then we do not need this cert validation.